src/Metadata/Property/Factory/SerializerPropertyMetadataFactory.php

```diff
@@ -29,12 +30,14 @@ final class SerializerPropertyMetadataFactory implements PropertyMetadataFactory
     private $resourceMetadataFactory;
     private $serializerClassMetadataFactory;
     private $decorated;
+    private $resourceClassResolver;
 
-    public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFactory, SerializerClassMetadataFactoryInterface $serializerClassMetadataFactory, PropertyMetadataFactoryInterface $decorated)
+    public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFactory, SerializerClassMetadataFactoryInterface $serializerClassMetadataFactory, PropertyMetadataFactoryInterface $decorated, ResourceClassResolverInterface $resourceClassResolver = null)
     {
         $this->resourceMetadataFactory = $resourceMetadataFactory;
         $this->serializerClassMetadataFactory = $serializerClassMetadataFactory;
         $this->decorated = $decorated;
+        $this->resourceClassResolver = $resourceClassResolver;
```

Comment:
Deprecated if `null`?

Reply:
Hmm... We need to do this in many places. Will do this in a follow-up PR?